geoh5py/objects/object_base.py

```diff
@@ -406,9 +406,7 @@ def create_property_group(
 
         return prop_group
 
-    def find_or_create_property_group(
-        self, name=None, uid=None, **kwargs
-    ) -> PropertyGroup:
+    def fetch_property_group(self, name=None, uid=None, **kwargs) -> PropertyGroup:
```

Comment:
Type the arguments, and the docstring doesn't really need  the `:obj:` thing since the return type is annotated.